File: src/main/java/xyz/oribuin/eternalreports/data/StaffPlayer.java
package xyz.oribuin.eternalreports.data;

import org.bukkit.entity.Player;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public class StaffPlayer {
    private final Player player;
    private final Map<UUID, Boolean> toggleMap;

    public StaffPlayer(Player player) {
        this.player = player;
        this.toggleMap = new HashMap<>();
    }

    public Player getPlayer() {
        return player;
    }

    public Map<UUID, Boolean> getToggleMap() {
        return toggleMap;
    }

    public boolean hasNotifications() {
        return  toggleMap.get(player.getUniqueId());
    }
}

File: src/main/kotlin/xyz/oribuin/eternalreports/data/StaffPlayer.kt
package xyz.oribuin.eternalreports.data

import org.bukkit.entity.Player
import java.util.*
import kotlin.collections.HashMap

class StaffPlayer(private val player: Player) {
    val toggleList: MutableMap<UUID, Boolean>

    init {
        toggleList = HashMap()
    }

    fun hasNotifications(): Boolean {
        return toggleList.getOrDefault(player.uniqueId, true)
    }
}